internal/configs/test_file.go

```diff
@@ -537,6 +662,154 @@ func decodeTestRunOptionsBlock(block *hcl.Block) (*TestRunOptions, hcl.Diagnosti
 	return &opts, diags
 }
 
+func decodeOverrideResourceBlock(block *hcl.Block, mode addrs.ResourceMode) (*OverrideResource, hcl.Diagnostics) {
+	parseTarget := func(attr *hcl.Attribute) (hcl.Traversal, *addrs.ConfigResource, hcl.Diagnostics) {
+		traversal, traversalDiags := hcl.AbsTraversalForExpr(attr.Expr)
+		diags := traversalDiags
+		if traversalDiags.HasErrors() {
+			return nil, nil, diags
+		}
+
+		configRes, configResDiags := addrs.ParseConfigResource(traversal)
+		diags = append(diags, configResDiags.ToHCL()...)
+		if configResDiags.HasErrors() {
+			return nil, nil, diags
+		}
+
+		return traversal, configRes, diags
+	}
+
+	res := &OverrideResource{
+		Mode: mode,
+	}
+
+	content, diags := block.Body.Content(overrideResourceBlockSchema)
+
+	if attr, exists := content.Attributes["target"]; exists {
+		target, parsed, moreDiags := parseTarget(attr)
+		res.Target, res.TargetParsed = target, parsed
+		diags = append(diags, moreDiags...)
+	}
+
+	if attr, exists := content.Attributes["values"]; exists {
+		v, moreDiags := parseObjectAttrWithNoVariables(attr)
+		res.Values, diags = v, append(diags, moreDiags...)
+	}
+
+	return res, diags
+}
+
+func decodeOverrideModuleBlock(block *hcl.Block) (*OverrideModule, hcl.Diagnostics) {
+	parseTarget := func(attr *hcl.Attribute) (hcl.Traversal, addrs.Module, hcl.Diagnostics) {
+		traversal, traversalDiags := hcl.AbsTraversalForExpr(attr.Expr)
+		diags := traversalDiags
+		if traversalDiags.HasErrors() {
+			return nil, nil, diags
+		}
+
+		target, targetDiags := addrs.ParseModule(traversal)
+		diags = append(diags, targetDiags.ToHCL()...)
+		if targetDiags.HasErrors() {
+			return nil, nil, diags
+		}
+
+		return traversal, target, diags
+	}
+
+	mod := &OverrideModule{}
+
+	content, diags := block.Body.Content(overrideModuleBlockSchema)
+
+	if attr, exists := content.Attributes["target"]; exists {
+		traversal, target, moreDiags := parseTarget(attr)
+		mod.Target, mod.TargetParsed = traversal, target
+		diags = append(diags, moreDiags...)
+	}
+
+	if attr, exists := content.Attributes["outputs"]; exists {
+		outputs, moreDiags := parseObjectAttrWithNoVariables(attr)
+		mod.Outputs, diags = outputs, append(diags, moreDiags...)
+	}
+
+	return mod, diags
+}
+
+func parseObjectAttrWithNoVariables(attr *hcl.Attribute) (map[string]cty.Value, hcl.Diagnostics) {
+	if vars := attr.Expr.Variables(); len(vars) != 0 {
```

Comment:
Is this explicitly needed? The Expr.Value(nil) below should produce a similar diagnostic?

Reply:
Agree, removed this call in c398e309df9d282bf765ba5578fef6573de032c0